### ISSUE TITLE
Ensure SA1512 (SingleLineCommentsMustNotBeFollowedByBlankLine) behaves with file headers

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#871]
The standard file header text for source files (see e.g. #839) uses line comments. The tests for SA1512 should include specific tests covering the behavior of this rule with respect to file headers.

Marking **needs discussion** because it's not clear to me at this time what the expected behavior is.

**Comment by vweijsters** [comment]
As far as I can determine, all consecutive single line comments at the start of the file (they may be separated by whitespace and blank lines) are considered the file header and therefore not subject to SA1512

I would propose we go a little bit further and say that all consecutive single line comments at the start of the file are considered the header. I think a blank line should signify the end of the header, I'm still undecided whether or not leading white space should be accepted.

**Comment by sharwell** [comment]
In the current implementation, is SA1512 reported if you include a space after the file header?

**Comment by vweijsters** [comment]
@sharwell I don't understand your question? You can't have a space after a single line comment afaik.

**Comment by sharwell** [comment]
I meant a blank line. Like this:

```csharp
// <copyright file="Widget.cs" company="Sprocket Enterprises">
//     Copyright (c) Sprocket Enterprises. All rights reserved.
// </copyright>

namespace NamespaceName
{
}
```

**Comment by vweijsters** [comment]
The current SA1512 implementation will produce a diagnostic for that, which is a bug.

The question is if it should produce a diagnostic if SA1633 (must have file header) is disabled?

**Comment by sharwell** [comment]
That's my preference as well, and I don't think [this line](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/517e3740d633ecd8d96f244176bc907afd4a6c89/StyleCop.Analyzers/StyleCop.Analyzers/AnalyzerExtensions.cs#L5) should produce a warning, which is the reason why SA1512 is temporarily disabled for this project.

I don't see any reason for SA1633 to impact this rule.

**Comment by vweijsters** [comment]
ok, I will fix this
